defaults/main.yml

```diff
@@ -51,27 +59,97 @@ postgres_container_volumemounts:
   - name: postgres-config
     mountPath: "/etc/secrets"
     readOnly: true
+  #- name: pg-archive
+  #  mountPath: /archive
 
 # Container limits
 postgres_container_limits:
   cpu: 800m
   # Note : to make things easier use the same unit as postgresql.conf (e.g. G, M, K etc...)
-  memory: 2G
+  memory: 600M
 
+# Master configuration
 postgres_config:
-  default_statistics_target : 100
-  maintenance_work_mem : 64MB
-  checkpoint_completion_target : 0.5
-  effective_cache_size : 1GB # Must be lower than container limits
-  work_mem : 4MB
-  wal_buffers : -1
-  checkpoint_segments : 3
-  shared_buffers : 128MB
+  default_statistics_target : "100"
+  maintenance_work_mem : "64MB"
+  checkpoint_completion_target : "0.5"
+  effective_cache_size : "1GB" # Must be lower than container limits
+  work_mem : "4MB"
+  wal_buffers : "-1"
+  checkpoint_segments : "3"
+  shared_buffers : "128MB"
+  wal_level: "hot_standby"
+  max_wal_senders: "5"
+  wal_keep_segments: "32"
+  archive_mode: "on"
 
 postgres_secrets:
   name: "postgres-config"
   type: "Opaque"
-  data:
-    password: "{{ pg_pass | b64encode }}"
-    postgresql.conf: "{{ lookup('template',role_path+'/templates/config/postgresql.conf') | b64encode }}"
-    pghba.conf: "{{ lookup('template',role_path+'/templates/config/pg_hba.conf') | b64encode }}"
+
+# Slaves
+# The following vars are used in master/slave cluster configuration
+postgres_slave_container_args:
+  - replication
+
+# List of slave servers,
+# the 'name' is randomly chosen. Can be anything
+# the 'nodeName' must be known by kubernetes
+shard_prefix: "slave"
+postgres_slave_shards:
+  - name: 1
+    nodeName: node3
+  - name: 2
+    nodeName: node4
+
+postgres_slave:
+  name: postgres-slave
+  replicas: 1
+  type: NodePort
+
+# Container limits
+postgres_slave_container_limits:
+  cpu: 800m
+  # Note : to make things easier use the same unit as postgresql.conf (e.g. G, M, K etc...)
+  memory: 600M
+
+postgres_slave_config:
+  default_statistics_target : "100"
+  maintenance_work_mem : "64MB"
+  checkpoint_completion_target : "0.5"
+  effective_cache_size : "512GB" # Must be lower than container limits
+  work_mem : "4MB"
+  wal_buffers : "-1"
+  checkpoint_segments : 3
+  shared_buffers : "256MB"
+  wal_level: "hot_standby"
+  max_wal_senders: "5"
+  wal_keep_segments: "32"
+  archive_mode: "on"
+  hot_standby: "on"
+
+postgres_slave_container_volumemounts:
+  - name: varlibpostgres
+    mountPath: /var/lib/postgresql/data
+  - name: postgres-config
+    mountPath: "/etc/secrets"
+    readOnly: true
+  #- name: pg-archive
+  #  mountPath: /archive
+  #  readOnly: true
+
+postgres_slave_volumes:
+    - name: varlibpostgres
+      hostPath:
+        path: "/containers/{{ domain_name|default('cluster.local') }}/rc-{{ postgres_slave.name }}/rootfs//var/lib/postgresql/data"
```

Comment:
It will be the same name for all shards. 
I don't know how to solve this. 
Maybe a inline replace after template is generated